subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -519,13 +558,19 @@ static bool broadcast_source_found(struct bt_data *data, void *user_data)
 	}
 
 	broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+	found_matched_broadcast_id = (broadcast_id == auto_scan.broadcast_id);
 
-	if (broadcast_id != auto_scan.broadcast_id) {
+	if (broadcast_name_scan.broadcast_name_is_matched) {
+		if (!strncmp(broadcast_name_scan.addr_str, addr_str, strlen(addr_str))) {
```

Comment:
I cant see the need for this check? You already know whether broadcast_name_is_matched

Reply:
But we need to find also the broadcast_id of the same address...

Reply:
Can you expand what you mean by other required parameters? When scanning specifically for a broadcast name, don't we only care about the name?

Reply:
According to my implementation, the `broadcast_id` is definitely needed. This check is used to find the `broadcast_id` corresponding to the same address as the matched `broadcast_name`. As for the other parameters, it was a mistake, and we do not need any other parameters besides `broadcast_id`. (I was considering handling some corner case scenarios, such as two broadcasts with the same name, which might require additional parameters, but they are not within the scope of this task.)